### ISSUE TITLE
Volumetric Fog Camera Preview changes as the user clicks on various nodes in the Scene Panel

### ISSUE THREAD

**Issue by golddotasksquestions** [godotengine/godot#62794]
### Godot version

4.0 Alpha11

### System information

Win  Nvidea 765M

### Issue description

I have also seen this in previous 4.0 versions, but I can't really reproduce reliably:
https://user-images.githubusercontent.com/47016402/177639962-a5c87f21-fe4a-4396-8618-0392b32dc9f5.mp4

### Steps to reproduce

1. Enable VolumetricFog in the WorldEnvironment
2. Enable Camera Preview
3. Click on nodes in the Scene Panel

### Minimal reproduction project

[volumetric_fog_cam_preview_bug.zip](https://github.com/godotengine/godot/files/9057954/volumetric_fog_cam_preview_bug.zip)

**Comment by golddotasksquestions** [comment]
@Calinou I suppose https://github.com/godotengine/godot/issues/62796 is related to https://github.com/godotengine/godot/issues/54892 as well?

**Comment by Calinou** [comment]
> @Calinou I suppose #62796 is then related to #54892 as well?

Yes, #62796 is due to the same cause as this issue, so I'll close it as a duplicate.

**Comment by golddotasksquestions** [comment]
@Calinou I suppose https://github.com/godotengine/godot/issues/62798 then is probably has the same cause as well and can be closed as duplicate?

**Comment by Calinou** [comment]
> @Calinou I suppose #62798 then is probably has the same cause as well and can be closed as duplicate?

Indeed.

**Comment by Mantle-Core** [comment]
> Related to #54892. This needs to be fixed in the same way, by forcing redrawing for a certain amount of frames whenever a volumetric fog parameter is changed or when a viewport is resized. The number of frames that need to be redrawn depends on the temporal reprojection amount chosen. With the default temporal reprojection amount of 0.9, you can reasonably assume that fog will be _mostly_ fully converged after 44 frames drawn (0.9 multiplied by 0.9 44 times <= 0.01).
> 
> Alternatively, temporal reprojection could be always disabled in the editor, but this will worsen volumetric fog quality in the editor.
> 
> This is effectively the opposite issue as #54893, but it happens only when you have no _visible_ FogVolume nodes in the scene.

@Calinou I took a shot at this issue. Would this be the expected output? At the moment this fix is for viewport resize.

https://github.com/godotengine/godot/assets/169231366/f01b86ee-b40f-4da0-930c-fab463b316ac

**Comment by Calinou** [comment]
@Mantle-Core Looks good, but make sure the editor isn't redrawing constantly for no reason. Enable **Show Update Spinner** in the Editor Settings and look at the top-right corner of the editor. The spinner should stop spinning when fog has fully converged.

**Comment by Mantle-Core** [comment]
The spinner stops after fully converging. I implemented it to force a redraw for N frames based on the Temporal Reprojection Amount. Afterwards, it stops redrawing.
